qa/acceptance-tests/src/test/java/io/camunda/it/client/ProcessDefinitionStatisticsTest.java

```diff
@@ -656,6 +784,22 @@ void shouldReturnStatisticsAndFilterByErrorHashCode() {
             .getProcessDefinitionKey();
     createInstance(processDefinitionKey);
     waitForProcessInstances(1, f -> f.processDefinitionKey(processDefinitionKey).hasIncident(true));
+    Awaitility.await("should receive data from ES")
+        .atMost(TIMEOUT_DATA_AVAILABILITY)
+        .ignoreExceptions() // Ignore exceptions and continue retrying
+        .untilAsserted(
+            () ->
+                assertThat(
+                        camundaClient
+                            .newIncidentSearchRequest()
+                            .filter(
+                                f ->
+                                    f.processDefinitionKey(processDefinitionKey)
+                                        .state(IncidentState.ACTIVE))
+                            .send()
+                            .join()
+                            .items())
+                    .hasSize(1));
```

Comment:
💭 If we don't yet have any `waitForIncidents` helper method like the `waitForProcessInstances` one above, we could add it now. I believe I have seen other places that wait for incidents as well.